Clickhouse 2.6.2 fails to cross-compile to ppc64le or win386.

See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18863 for original report.

Clickhouse fails to compile cross-platform for ppc64le or win386.

```
cd ./cmd/otelcontribcol && GO111MODULE=on CGO_ENABLED=0 go build -trimpath -o ../../bin/otelcontribcol_linux_ppc64le \
	-ldflags "-X github.com/open-telemetry/opentelemetry-collector-contrib/internal/otelcontribcore/internal/version.Version=9d0020e" -tags "" .
# github.com/ClickHouse/clickhouse-go/v2/lib/column
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:100:40: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:116:42: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:142:31: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/fixed_string.go:147:33: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
Error: /home/runner/go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.6.2/lib/column/string.go:74:50: too many arguments in call to binary.Str2Bytes
	have (string, int)
	want (string)
make: *** [Makefile:259: otelcontribcol] Error 2
Error: Process completed with exit code 2.
```

--- a/lib/binary/string_safe.go
+++ b/lib/binary/string_safe.go
@@ -20,6 +20,13 @@
 
 package binary
 
-func Str2Bytes(str string) []byte {
-	return []byte(str)
+func Str2Bytes(str string, expectedLen int) []byte {
+	b := []byte(str)
+	if len(str) < expectedLen {
+		extended := make([]byte, expectedLen)
+		copy(extended, b)
+		return extended
+	}
+
+	return b
 }

